### ISSUE TITLE
Admin account not recognized on new installation

### ISSUE THREAD

**Issue by wakerb** [processone/ejabberd#1255]
> What version of ejabberd are you using?

I tried both 16.08 and 16.06

> What operating system (version) are you using?

Windows Server 2008R2, Windows 7 and 10

> How did you install ejabberd (source, package, distribution)?

Windows Installer downloaded from Process One.

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

After installing ejabberd, and attempting to log into the admin interface the login credentials are not recognized.  The system thinks the default admin account does not exist. The install and ejabberd logs are attached.
[ejabberd-logs.zip](https://github.com/processone/ejabberd/files/423073/ejabberd-logs.zip)

**Comment by badlop** [comment]
I've tested using ejabberd 16.08 linux compiled from source, configured in ejabberd.yml the host "Jabber":
```
hosts:
  - "Jabber"
```
Then I created account admin@Jabber and logged in it.
```
11:13:01.872 [info] ({socket_state,gen_tcp,#Port<0.22734>,<0.1339.0>})
 Accepted legacy authentication for admin@jabber/tka1 by ejabberd_auth_mnesia from 127.0.0.1
```
Then in the webadmin, provided as user "admin@Jabber", and the proper password, and I was able to login.

If I provide another username or another servername in the login window, then it returns the same error you get:
```
11:14:32.789 [warning] Access of <<"admin2@Jabber">> from <<"127.0.0.1">> failed with error:
 <<"inexistent-account">>
```

In your case, the problem is that the installer is not able to register the account:
```
C:\Program Files\ejabberd-16.06\bin>bash ejabberdctl register "admin" "Jabber" "12345"
Can't register user admin@Jabber at node ejabberd@localhost: not_allowed
```

This is probably a bug in the installer. Until this is fixed, you can enable mod_register in the configuration file, then create that account yourself using a Jabber client, or calling the command line:
> ejabberdctl register "admin" "Jabber" "12345"

**Comment by debalance** [comment]
I guess this is because of #1231, at least with 16.08.

**Comment by wakerb** [comment]
Thanks for your help guys, I've been away for a few days.

Badlop, I'm not quite sure what you mean when you say "enable mod_register"
I found the configuration file. The mod_register section, according to the documentation, contains a number of attributes. I'm not quite sure what to change.  Below is the mod_register section from my configuration file.

  mod_register:
    ##
    ## Protect In-Band account registrations with CAPTCHA.
    ##
    ##   captcha_protected: true
    ##
    ## Set the minimum informational entropy for passwords.
    ##
    ##   password_strength: 32
    ##
    ## After successful registration, the user receives
    ## a message with this subject and body.
    ##
    welcome_message:
      subject: "Welcome!"
      body: |-
        Hi.
        Welcome to this XMPP server.
    ##
    ## When a user registers, send a notification to
    ## these XMPP accounts.
    ##
    ##   registration_watchers:
    ##     - "admin1@example.org"
    ##
    ## Only clients in the server machine can register accounts
    ##
    ip_access: trusted_network
    ##
    ## Local c2s or remote s2s users cannot register accounts
    ##
    ##   access_from: deny
    access: register

**Comment by badlop** [comment]
Apparently you have mod_register already enabled. So, continue with my steps: try to register the admin account, then try again to login in the web.

**Comment by wakerb** [comment]
I did use Miranda IM client and did appear to create an admin user (I think).  A small IM style window did pop up, but showed admin@jabber to be offline.  I am still unable to connect to the admin interface. I can not connect to this machine from any other machine with any type of client.
  
I downloaded and installed Openfire Server on the same machine.  It just worked. I created a bunch of tables in my SQL database with very little effort and can establish chat sessions on demand with several clients from any machine. I can now proceed with the IM component of my project. 

It seems to me that on Windows, Ejabberd is like a nice, feature loaded, high performance car that just won’t start.  In other words, it’s useless.
